log4j-scan.py

```diff
@@ -265,49 +284,51 @@ def scan_url(url, callback_host):
     payloads = [payload]
     if args.waf_bypass_payloads:
         payloads.extend(generate_waf_bypass_payloads(f'{parsed_url["host"]}.{callback_host}', random_string))
+
     for payload in payloads:
         cprint(f"[•] URL: {url} | PAYLOAD: {payload}", "cyan")
-        if args.request_type.upper() == "GET" or args.run_all_tests:
-            try:
-                requests.request(url=url,
-                                 method="GET",
-                                 params={"v": payload},
-                                 headers=get_fuzzing_headers(payload),
-                                 verify=False,
-                                 timeout=timeout,
-                                 allow_redirects=(not args.disable_redirects),
-                                 proxies=proxies)
-            except Exception as e:
-                cprint(f"EXCEPTION: {e}")
-
-        if args.request_type.upper() == "POST" or args.run_all_tests:
-            try:
-                # Post body
-                requests.request(url=url,
-                                 method="POST",
-                                 params={"v": payload},
-                                 headers=get_fuzzing_headers(payload),
-                                 data=get_fuzzing_post_data(payload),
-                                 verify=False,
-                                 timeout=timeout,
-                                 allow_redirects=(not args.disable_redirects),
-                                 proxies=proxies)
-            except Exception as e:
-                cprint(f"EXCEPTION: {e}")
-
-            try:
-                # JSON body
-                requests.request(url=url,
-                                 method="POST",
-                                 params={"v": payload},
-                                 headers=get_fuzzing_headers(payload),
-                                 json=get_fuzzing_post_data(payload),
-                                 verify=False,
-                                 timeout=timeout,
-                                 allow_redirects=(not args.disable_redirects),
-                                 proxies=proxies)
-            except Exception as e:
-                cprint(f"EXCEPTION: {e}")
+        for param in args.target_parameters:
```

Comment:
This will result in dozens of requests being sent, while it can be sent in the same request, can you please add them into the params dict as a single request?

Reply:
That's better. I did the changes you requested. Now you anyone can provide custom parameters as follows,
```
'param1=value1&param2={{inject}}&param3={{inject}}'
```
They can specify to which parameters they want to inject (This will be helpful in some situations)